llama_cpp/server/model.py

```diff
@@ -44,12 +57,49 @@ def __call__(self, model: Optional[str] = None) -> llama_cpp.Llama:
             if self._current_model is not None:
                 return self._current_model
 
+        new_settings = self._model_settings_dict[model]
+        
+        if self._current_model is not None and self._current_model_alias is not None:
+            current_settings = self._model_settings_dict[self._current_model_alias]
+
+            def hot_swappable_settings(settings: ModelSettings) -> Dict[str, Any]:
+                """Subset of settings used to check if models can be hot-swapped"""
+                values = settings.model_dump()
+                values.pop('model_alias', None) # The model alias doesn't matter
+                values.pop('lora_adapters', None) # Different LoRA adapters can be hot-swapped
+                return values
+            
+            if hot_swappable_settings(new_settings) == hot_swappable_settings(current_settings):
```

Comment:
This fast path lets us avoid loading the model again if it is the same model. If we are able to hot-swap then we can just update the LoRA adapter scales (loading new LoRAs if needed) and then exit early.